content/blog/pulumi-components/index.md

```diff
@@ -0,0 +1,339 @@
+
+1. **Cross-language consumption**: Components authored in one language can now be consumed in any Pulumi language, including Pulumi YAML.
+
+2. **Simplified sharing**: Share components by simply pushing code to a git repository - no need to manually publish SDKs.
+
+3. **Streamlined consumption**: Use the `pulumi package add` command with any git URL (like `github.com/org/repo`) to easily incorporate any components into your projects.
+
+## How It Works
+
+### Creating Components
+
+Creating a component remains the same as before—define a class that extends `ComponentResource`:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+export interface SecureBucketArgs {
+    bucketName?: pulumi.Input<string>;
+    versioning?: boolean;
+    encryption?: boolean;
+    tags?: { [key: string]: pulumi.Input<string> };
+}
+
+export class SecureBucket extends pulumi.ComponentResource {
+    public readonly bucket: aws.s3.Bucket;
+    public readonly bucketName: pulumi.Output<string>;
+
+    constructor(name: string, args: SecureBucketArgs = {}, opts?: pulumi.ComponentResourceOptions) {
+        super("my:components:SecureBucket", name, {}, opts);
+
+        // Create an S3 bucket with best practices by default
+        this.bucket = new aws.s3.Bucket(`${name}`, {
+            bucket: args.bucketName,
+            tags: {
+                ManagedBy: "Pulumi",
+                ...args.tags,
+            },
+        }, { parent: this });
+
+        // Conditionally enable versioning
+        if (args.versioning !== false) {
+            new aws.s3.BucketVersioningV2(`${name}-versioning`, {
+                bucket: this.bucket.id,
+                versioningConfiguration: {
+                    status: "Enabled",
+                },
+            }, { parent: this });
+        }
+
+        // Conditionally enable encryption
+        if (args.encryption !== false) {
+            new aws.s3.BucketServerSideEncryptionConfigurationV2(`${name}-encryption`, {
+                bucket: this.bucket.id,
+                rules: [{
+                    applyServerSideEncryptionByDefault: {
+                        sseAlgorithm: "AES256",
+                    },
+                }],
+            }, { parent: this });
+        }
+
+        this.bucketName = this.bucket.id;
+
+        this.registerOutputs({
+            bucket: this.bucket,
+            bucketName: this.bucketName,
+        });
+    }
+}
+```
+
+### Sharing Components
+
+Pulumi's new Components capability is designed to make sharing infrastructure building blocks as easy as possible, whether you're publishing them to a Git repository or referencing them from a local folder in a monorepo.
+
+#### Publishing to Git
+
+Once you've authored your component, sharing it is as simple as pushing your code to a Git repository and optionally tagging a release. From there, anyone (including you) can pull it down and use it in their own Pulumi projects.
```

Comment:
We never mention the need for `PulumiPlugin.yaml`. Maybe not super important for the blog, just making sure that this is intended and we don't think it's slightly deceptive/confusing.